File: 06 - DAT203.1x - Data Science Essentials/Labs/Mod5/Sample Code/FilterOutliers.R
id.outlier <- function(df){
  ## Use ifelse to filter for outliers
  df[, "outlier"] <- ifelse(df[,"enginesize"] > 190 | 
                              df[, "weight"] > 3500| 
                              df[, "citympg"] > 40, '1', '0')   
  df
}

## ID and plot outliers
auto.price <- maml.mapInputPort(1) # read autos data frame from port 1
require(dplyr) # make sure dplyr is loaded
auto.price <- id.outlier(auto.price) # mark the outliers
## remove rows where outlier is 1, then remove outlier column
out <- auto.price %>% filter(outlier == 0) %>% subset(select=-outlier) # filter outliers
maml.mapOutputPort("out") ## Output the data frame

File: 06 - DAT203.1x - Data Science Essentials/Labs/Mod5/Sample Code/FilterOutliers.py
def id_outlier(df):
    ## Create a vector of 0 of length equal to the number of rows
    temp = [0] * df.shape[0]
    ## test each outlier condition and mark with a 1 as required
    for i, x in enumerate(df['enginesize']):
        if (x > 190): temp[i] = 1 
    for i, x in enumerate(df['weight']):
        if (x > 3500): temp[i] = 1 
    for i, x in enumerate(df['citympg']):
        if (x > 40): temp[i] = 1      
    df['outlier'] = temp # append a column to the data frame
    return df


def azureml_main(df): 
    ## Define the plot columns
    df = id_outlier(df)  # mark outliers       
    df = df[df.outlier == 0] # filter for outliers
    df.drop('outlier', axis = 1, inplace = True)
    return df
